aten/src/ATen/native/cuda/IndexKernel.cu

```diff
@@ -385,20 +381,9 @@ void masked_scatter_cuda_impl(Tensor& self, const Tensor& mask, const Tensor& so
   // Use a prefix sum to determine the output locations of the masked elements
```

Comment:
Can you please also move `totalElements` calculation here (line 373), and use direct cub calls, since `mask` is made contiguous anyway. Turns out mask.sum() first converts mask to long, and then reduces long tensor. Using cub directly should be much better.

Reply:
done